tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -18,24 +19,25 @@ internal static class DatabaseMonitoringPropagator
         private const string SqlCommentEnv = "dde";
         internal const string DbmPrefix = $"/*{SqlCommentSpanService}='";
 
-        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, SpanContext context, IntegrationId integrationId, out bool traceParentInjected)
+        internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, string configuredServiceName, Span span, IntegrationId integrationId, out bool traceParentInjected)
         {
             traceParentInjected = false;
 
             if ((integrationId is IntegrationId.MySql or IntegrationId.Npgsql or IntegrationId.SqlClient) &&
                 (propagationStyle is DbmPropagationLevel.Service or DbmPropagationLevel.Full))
             {
                 var propagatorStringBuilder = StringBuilderCache.Acquire(StringBuilderCache.MaxBuilderSize);
-                propagatorStringBuilder.Append(DbmPrefix).Append(Uri.EscapeDataString(context.ServiceNameInternal)).Append('\'');
+                var dddbs = (span.Tags is SqlV1Tags sqlTags) ? sqlTags.PeerService : span.Context.ServiceNameInternal;
```

Comment:
The fact that `peer.service` can be an override, the `dbname` or `out.host` is wanted I assume. But asking just in case.

Reply:
As read in [the original doc](https://docs.google.com/document/d/1VjHPLw9oGnMhjQxCS1JDNz3oQxg9AdjWKMVZmEgCF6Y)

`The correct value is whatever we would expose to the customer on the trace flamegraph and dependency map`

Which is not a very technical term, but in this case means whatever is set in `peer.service`